server/controllers/voteController.js

```diff
@@ -0,0 +1,104 @@
+import Post from '../models/postModel.js';
+import Vote from '../models/voteModel.js';
+
+
+const votePost = async (req, res) => {
+
+    const { postId, userId, voteOption } = req.body;
+
+    if (userId !== req.user._id.toString())
+        return res.status(500).send({ message: "something went wrong" });
+
+    try {
+        const post = await Post.findOne({ '_id': postId });
+
+        if (!post)
+            return res.status(404).send({ message: "Post was not found" });
+
+        const voteExists = await Vote.findOne({ postId });
+
+        if (voteExists) {
+
+            if (voteOption === 'upvote') {
+
+                if (!await Vote.findOne({ postId, userId: { $elemMatch: { userId } } })) {
```

Comment:
what does this mean ? the creator of post can't vote up his own posts ?
just asking out of curiosity

Reply:
he can, if you talking about line 24 then i am checking if the user already exists in the collection if not i am adding him